### ISSUE TITLE
pyre2 performance 3 times slower

### ISSUE THREAD

**Issue by andresriancho** [axiak/pyre2#12]
I've been playing around with pyre2 for a while and implemented it into w3af [0]. During this experiment I run the performance tests that come with pyre2 and was amazed with its speed! Then I created a new branch [1] and compared the results of running some of our unit-tests which are very dependent on regular expressions; where I was very surprised to see these results:

With pyre2 [1]:

``` bash
dz0@dz0-laptop:~/workspace/pyre2$ time nosetests --with-doctest plugins/grep/
.....re2/dfa.cc:447: DFA out of memory: prog size 95874 mem 2412486
................
----------------------------------------------------------------------
Ran 21 tests in 29.475s

OK

real    0m31.980s
user    0m26.150s
sys 0m0.570s
```

With regular "import re" [2]:

``` bash
dz0@dz0-laptop:~/workspace/w3af$ time nosetests --with-doctest plugins/grep/
.....................
----------------------------------------------------------------------
Ran 21 tests in 8.093s

OK

real    0m9.971s
user    0m4.550s
sys 0m0.590s
```

To sum up, pyre2 uses ~30seg to run these tests, while re uses ~10seg.

I've been reading the documentation and it seems that this might be because of unicode strings being passed to pyre2; could you confirm? If so... when are you planning to fix this issue? Is it even fixable? With py3k where everything is a unicode string... this issue will become a huge impediment for future pyre2 users!

Let me know if there is something I can do to help with testing, I would love to have pyre2's performance enhancements in my project's code!

[0] http://w3af.org/
[1] https://w3af.svn.sourceforge.net/svnroot/w3af/branches/pyre2
[2] https://w3af.svn.sourceforge.net/svnroot/w3af/trunk

**Comment by axiak** [comment]
Try running with more memory
